Include/internal/pycore_frame.h

```diff
@@ -20,13 +20,13 @@ enum _framestate {
 typedef signed char PyFrameState;
 
 typedef struct _interpreter_frame {
-    PyObject *f_globals;
-    PyObject *f_builtins;
-    PyObject *f_locals;
-    PyCodeObject *f_code;
-    PyFrameObject *frame_obj;
-    /* Borrowed reference to a generator, or NULL */
-    PyObject *generator;
+    PyFunctionObject *f_func; /* Strong reference */
```

Comment:
Couldn't this just as well be `PyFrameConstructor *`?  Then we wouldn't need to create temporary function objects in the various places (and this PR has much less churn).

I expect it's due to the refcount, both of the underlying function object itself and for the borrowed references (e.g. `f_globals`).  This implies the code using the frame might not hold a reference to the function (nor does its caller).  However, that doesn't seem right.  If that's not the case then why do we need to keep the function here instead of `PyFrameConstructor *`?

Reply:
We need a strong reference to the function to keep the globals and builtins alive.

Reply:
My point is:  don't we have a guarantee that the code that created/initialized the frame holds a reference to the function object (or to the globals/builtins objects)?

Reply:
We don't care who else has a reference. There is one here. That is enough.

Reply:
If we know other code always has a reference to the function then there's no need for the frame to keep a strong reference. 

(FWIW, I brought this up because it seams like in this PR we're trying to reduce the number of incref/decref we do relative to frame lifecycle.  If not then why bother with borrowed references for any of the fields?)

Reply:
There may not be a reference to the function anywhere:
```
(lambda x: x)(1)
```
There is no reference to the lambda from the caller when it is being executed.